metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/utils/EntityDao.java

```diff
@@ -0,0 +1,263 @@
+import eu.europeana.enrichment.api.internal.PlaceTermList;
+import eu.europeana.enrichment.api.internal.TimespanTermList;
+import eu.europeana.metis.utils.ExternalRequestUtil;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Simon Tzanakis
+ * @since 2020-07-07
+ */
+public class EntityDao {
+
+  private final AdvancedDatastore datastore;
+
+  public static final String ID_FIELD = "_id";
+  private static final String ENTITY_TYPE_FIELD = "entityType";
+  public static final String CODE_URI_FIELD = "codeUri";
+  private static final String MODIFIED_FIELD = "modified";
+  private static final String OWL_SAME_AS = "owlSameAs";
+
+  //Entity types
+  private static final String CONCEPT_TYPE = "ConceptImpl";
+  private static final String PLACE_TYPE = "PlaceImpl";
+  private static final String AGENT_TYPE = "AgentImpl";
+  private static final String TIMESPAN_TYPE = "TimespanImpl";
+  public static final String ORGANIZATION_TYPE = "OrganizationImpl";
+
+  //Table names
+  private static final String AGENT_TABLE = "people";
+  private static final String CONCEPT_TABLE = "concept";
+  private static final String PLACE_TABLE = "place";
+  private static final String TIMESPAN_TABLE = "period";
+  public static final String ORGANIZATION_TABLE = "organization";
+  private final MongoClient mongoClient;
+
+
+  public EntityDao(MongoClient mongoClient, String databaseName) {
+    this.mongoClient = mongoClient;
+
+    // Register the mappings and set up the data store.
+    final Morphia morphia = new Morphia();
+    morphia.map(MongoTermList.class);
+    morphia.map(AgentTermList.class);
+    morphia.map(ConceptTermList.class);
+    morphia.map(PlaceTermList.class);
+    morphia.map(TimespanTermList.class);
+    morphia.map(OrganizationTermList.class);
+    this.datastore = (AdvancedDatastore) morphia.createDatastore(this.mongoClient, databaseName);
+  }
+
+  public <T extends MongoTermList<S>, S extends AbstractEdmEntityImpl> MongoTermList<S> findTermListByField(
+      Class<T> mongoTermListType, String fieldName, String fieldValue) {
+    return ExternalRequestUtil.retryableExternalRequestConnectionReset(
+            () -> this.datastore.find(mongoTermListType).filter(fieldName, fieldValue).first());
+  }
+
+
+  private MongoTerm findMongoTermByField(String entityType, String fieldName, String fieldValue) {
+    return ExternalRequestUtil.retryableExternalRequestConnectionReset(
+        () -> this.datastore.find(entityType, MongoTerm.class).filter(fieldName, fieldValue)
+            .first());
+  }
+
+  public List<MongoTerm> getAllMongoTerms(EntityType entityType) {
+    Query<MongoTerm> query = this.datastore.createQuery(getTableName(entityType), MongoTerm.class);
```

Comment:
createQuery is deprecated.

Reply:
Yes there is no other way at the moment, we'll need to change the db structure to fix this